src/sentry/sentry_apps/utils/errors.py

```diff
@@ -1,49 +1,47 @@
+from collections.abc import Mapping, Sequence
 from enum import Enum
 
+from sentry.utils import json
+
 
 class SentryAppErrorType(Enum):
     CLIENT = "client"
     INTEGRATOR = "integrator"
     SENTRY = "sentry"
 
 
-# Represents a user/client error that occured during a Sentry App process
-class SentryAppError(Exception):
-    error_type = SentryAppErrorType.CLIENT
-    status_code = 400
+class SentryAppBaseError(Exception):
+    error_type: SentryAppErrorType
+    status_code: int
 
     def __init__(
         self,
         error: Exception | None = None,
         status_code: int | None = None,
     ) -> None:
-        if status_code:
-            self.status_code = status_code
+        self.status_code = status_code or self.status_code
+        self.error = error
+
+    def __str__(self) -> str:
+        if isinstance((Mapping, Sequence), self.error.args[0]):
+            return json.dumps(self.error.args[0])
+        else:
+            return str(self.error.args[0])
```

Comment:
you can also `json.dumps` a string, it will just wrap it in double quotes

Reply:
if we `json.dumps` a string tho we'll get `"'lmao'"` from `'lmao'` , wouldn't that be more confusing of an output?